src/main/java/seedu/address/logic/commands/EditProjectCommand.java

```diff
@@ -0,0 +1,211 @@
+        assert projectToEdit != null;
+
+        ProjectId projectId = projectToEdit.getId();
+        ProjectName updatedName = editProjectDescriptor.getName().orElse(projectToEdit.getName());
+        Set<Skill> updatedSkills = editProjectDescriptor.getSkills().orElse(projectToEdit.getSkills());
+
+        return new Project(updatedName, projectId, updatedSkills);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof EditProjectCommand)) {
+            return false;
+        }
+
+        EditProjectCommand otherEditProjectCommand = (EditProjectCommand) other;
+        return index.equals(otherEditProjectCommand.index)
+                && editProjectDescriptor.equals(otherEditProjectCommand.editProjectDescriptor);
+    }
+
+    @Override
+    public String toString() {
+        return new ToStringBuilder(this)
+                .add("index", index)
+                .add("editProjectDescriptor", editProjectDescriptor)
+                .toString();
+    }
+
+    /**
+     * Stores the details to edit the project with. Each non-empty field value will
+     * replace the corresponding field value of the project.
+     */
+    public static class EditProjectDescriptor {
+        private ProjectName name;
+        private Set<Skill> skills;
+
+        public EditProjectDescriptor() {
+        }
+
+        /**
+         * Copy constructor.
+         * A defensive copy of {@code tags} is used internally.
+         */
+        public EditProjectDescriptor(EditProjectDescriptor toCopy) {
+            setName(toCopy.name);
+            setSkills(toCopy.skills);
+        }
+
+        /**
+         * Returns true if at least one editable field is edited.
+         */
+        public boolean isAnyFieldEdited() {
+            return CollectionUtil.isAnyNonNull(name, skills);
+        }
+
+        public void setName(ProjectName name) {
+            this.name = name;
+        }
+
+        public Optional<ProjectName> getName() {
+            return Optional.ofNullable(name);
+        }
+
+        /**
+         * Sets {@code skills} to this object's {@code skills}.
+         * A defensive copy of {@code skills} is used internally.
+         */
+        public void setSkills(Set<Skill> skills) {
+            this.skills = (skills != null) ? new HashSet<>(skills) : null;
+        }
+
+        /**
+         * Returns an unmodifiable skill set, which throws
+         * {@code UnsupportedOperationException} if modification is attempted.
+         * Returns {@code Optional#empty()} if {@code skills} is null.
+         */
+        public Optional<Set<Skill>> getSkills() {
+            return (skills != null) ? Optional.of(Collections.unmodifiableSet(skills)) : Optional.empty();
+        }
+
+        @Override
+        public boolean equals(Object other) {
```

Comment:
Though general test coverage is good, these parts are not covered. Tests could be added for even better coverage?